package/MDAnalysis/analysis/hbond_analysis.py

```diff
@@ -0,0 +1,199 @@
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""Hydrogen Bond Analysis --- :mod:`MDAnalysis.analysis.hbond_analysis`
+===========================================================================
+
+:Author: Paul Smith and Mateusz Bieniek
+:Year: 2019
+:Copyright: GNU Public License v3
+
+This module provides methods to find and analyse hydrogen bonds in a Universe.
+
+The :class:`HydrogenBondAnalysis` class is a new version of the original :class:`HydrogenBondAnalysis`
+class from the module mod:`MDAnalysis.analysis.hbonds.hbond_analysis`, which itself was modeled after the `VMD
+HBONDS plugin`_.
+
+.. _`VMD HBONDS plugin`: http://www.ks.uiuc.edu/Research/vmd/plugins/hbonds/
+
+
+Options:
+  - *d_h_cutoff* (Å) [1.2] : Distance cutoff used for finding donor-hydrogen pairs
+  - *d_a_cutoff* (Å) [3.0] : Distance cutoff for hydrogen bonds. This cutoff refers to the D-A distance.
+  - *d_h_a_angle_cutoff* (degrees) [150] : D-H-A angle cutoff for hydrogen bonds.
+
+
+Output:
+  - *frame* : frame at which a hydrogen bond was found
+  - *donor id* : atom id of the hydrogen bond donor atom
+  - *hydrogen id* : atom id of the hydrogen bond hydrogen atom
+  - *acceptor id* : atom id of the hydrogen bond acceptor atom
+  - *distance* (Å): length of the hydrogen bond
+  - *angle* (degrees): angle of the hydrogen bond
+
+Hydrogen bond data are returned in a :class:`numpy.ndarray` on a "one line, one observation" basis
+and can be accessed via :attr:`HydrogenBondAnalysis.hbonds`::
+
+    results = [
+        [
+            <frame>,
+            <donor index (0-based)>,
+            <hydrogen index (0-based)>,
+            <acceptor index (0-based)>,
+            <distance>,
+            <angle>
+        ],
+        ...
+    ]
+
+Classes
+-------
+
+.. autoclass:: HydrogenBondAnalysis
+   :members:
+
+   .. attribute:: hbonds
+
+"""
+
+import numpy as  np
+
+from . import base
+from MDAnalysis.lib.distances import capped_distance
+from MDAnalysis.lib.distances import calc_angles
+
+class HydrogenBondAnalysis(base.AnalysisBase):
+    """
+    Perform an analysis of hydrogen bonds in a Universe.
+    """
+
+    def __init__(self, universe, donors_sel, hydrogens_sel, acceptors_sel, d_h_cutoff, d_a_cutoff, d_h_a_angle_cutoff):
```

Comment:
Can the call signature be made the same as for the original class?

Reply:
It can, but to be honest I find the call signature of the current class very confusing. When I first tried using current analysis tool I was put off by the large number of selections I had to provide and especially by the idea of having to create a new class if I wanted different donor or acceptor types. I actually preferred the idea of writing my own hydrogen bond analysis script, which forms the basis of the new class.

I know there are now `donor` and `acceptor` keywords that can be used to add atom names not already included in the default selections, but there is no `hydrogens` keyword, meaning that a new class must be created if someone is using a different forcefield or if they are using LAMMPS (which has numbered atom types).

I find using only `donors_sel`, `acceptors_sel` and `hydrogens_sel` much cleaner and easier to understand than the various combinations provided by `selection1`, `selection2`, `selection1_type`, `forcefield`, `donors` and `acceptors`. The other issue with the current default selections is that they only take into account protein atom names and not lipid or CGenFF names. 

I'm not sure how you feel about this, but if rewriting the class involves changing how the results are stored/returned then it might also be a good idea to simplify the signature call.